packages/ag-charts-community/src/chart/series/hierarchy/hierarchySeries.test.ts

```diff
@@ -0,0 +1,113 @@
+import type { BBox } from '../../../scene/bbox';
+import type { ChartAxisDirection } from '../../chartAxisDirection';
+import type { ChartLegendDatum, ChartLegendType } from '../../legendDatum';
+import type { SeriesNodeDataContext } from '../series';
+import type { SeriesTooltip } from '../seriesTooltip';
+import { HierarchySeries } from './hierarchySeries';
+
+class ExampleHierarchySeries extends HierarchySeries<any> {
+    override tooltip: SeriesTooltip<any> = null!;
+
+    override getSeriesDomain(_direction: ChartAxisDirection): any[] {
+        throw new Error('Method not implemented.');
+    }
+
+    override createNodeData(): Promise<SeriesNodeDataContext<any, any>[]> {
+        throw new Error('Method not implemented.');
+    }
+
+    override update(_opts: { seriesRect?: BBox | undefined }): Promise<void> {
+        throw new Error('Method not implemented.');
+    }
+
+    override getTooltipHtml(_seriesDatum: any): string {
+        throw new Error('Method not implemented.');
+    }
+
+    override getLegendData(_legendType: unknown): ChartLegendDatum<any>[] | ChartLegendDatum<ChartLegendType>[] {
+        throw new Error('Method not implemented.');
+    }
+}
+
+describe('hierarchySeries', () => {
+    it('Creates a hierarchy', async () => {
+        const series = new ExampleHierarchySeries(null!);
+        series.sizeKey = 'size';
+        series.data = [
+            { size: 5, children: [{ size: 1 }, { size: 2 }, { size: 3 }] },
+            {
+                size: 5,
+                children: [
+                    { size: 1 },
+                    { size: 2, children: [{ size: 4 }, { size: 5 }, { size: 6 }] },
+                    { size: 3, children: [{ size: 7 }] },
+                ],
+            },
+        ];
+        await series.processData();
+
+        series.rootNode.walk((node) => {
+            // @ts-expect-error
+            delete node.series;
+            delete node.datum;
+            delete node.parent;
+        });
+
+        expect(series.rootNode).toMatchSnapshot();
+        expect(series.rootNode.sumSize).toBe(5 + 1 + 2 + 3 + 5 + 1 + 2 + 4 + 5 + 6 + 3 + 7);
+        expect(series.maxDepth).toBe(2);
+    });
+
+    it('Handles an empty dataset', async () => {
+        const series = new ExampleHierarchySeries(null!);
+        series.data = [];
+        await series.processData();
+
+        // @ts-expect-error - Jest will complain about circular dependencies otherwise
```

Comment:
Likewise, what is the root cause of this circular dep - if Jest is complaining about it, maybe it will be a real issue that will bite us later?

Reply:
Updated comment to

> @ts-expect-error - Remove circular dependencies because if this test fails, Jest won't be able to print any errors

The hierarchy node has to have `series` as a property to conform to `SeriesNodeDatum`. And the series contains the rootNode, so it is a circular reference